packages/skill-lesson/trpc/routers/invoices.ts

```diff
@@ -0,0 +1,67 @@
+import {z} from 'zod'
+import {publicProcedure, router} from '../trpc.server'
+import {getSdk} from '@skillrecordings/database'
+import {stripe} from '@skillrecordings/commerce-server'
+
+export const invoicesRouter = router({
```

Comment:
tRPC router for the invoice page, it fetches all the relevant records (charge, purchase, bulk coupon) and then determines the quantity with logic for when to fall back to the bulk coupon's quantity.